packages/babel-plugin-transform-exponentiation-operator/src/index.ts

```diff
@@ -1,26 +1,92 @@
 import { declare } from "@babel/helper-plugin-utils";
-import build from "@babel/helper-builder-binary-assignment-operator-visitor";
-import { types as t } from "@babel/core";
+import type { types as t, Scope } from "@babel/core";
 
 export default declare(api => {
   api.assertVersion(REQUIRED_VERSION(7));
 
+  const { types: t } = api;
+
+  function build(left: t.Expression, right: t.Expression) {
+    return t.callExpression(
+      t.memberExpression(t.identifier("Math"), t.identifier("pow")),
+      [left, right],
+    );
+  }
+
+  function maybeMemoize<T extends t.Expression | t.Super>(
+    node: T,
+    scope: Scope,
+  ) {
+    if (scope.isPure(node) || t.isSuper(node)) {
```

Comment:
In the nullish transform we use `scope.isStatic` for the memoization purpose:

https://github.com/babel/babel/blob/b01932768b978bcaa4aa0517d4bdba29ff3966f8/packages/babel-plugin-transform-nullish-coalescing-operator/src/index.ts#L28-L41

Can you also add a new test and see if we can handle the edge case `function (a, b = a.b.c ** 2) {}` here?